app/frontend/users/UserEdit.res

```diff
@@ -48,13 +50,20 @@ type action =
   | StartDeletingAccount
   | FinishAccountDeletion
   | UpdateEmailAndDisableInput(string)
+  | CancelNewEmailRequest(string)
+
+let newValue = switch Dom.Storage2.getItem(Dom.Storage2.localStorage, "newEmail") {
```

Comment:
It's unclear why we need to use the browser _localStorage_ to store this value. This is also not in sync with the value in the server that is stored in the `new_email` column in the users table. 

Also `localStorage` is specific to a browser. Imagine a scenario where you create a request in one browser and later visit the user edit page from a different system/browser. The behavior will be different even though you have a valid email change request live.

The safest method is to always pull in the data from the server available in the `new_email` column and store it in state in an option<string> field.